### DISCUSSION TITLE
Magefiles in their own nested Go module

### DISCUSSION THREAD

**Discussion by yann-soubeyrand** [magefile/mage#464] [Q&A] [answered]
Hello,

I’m trying to put my Magefiles in their own Go module to not mix my application dependencies and my Magefiles dependencies. Here is what it looks like:
```
root
├─ go.mod
├─ go.sum
├─ *.go
└─ magefiles
   ├─ go.mod
   ├─ go.sum
   └─ *.go
```
However, I can’t use the `mage` command at the root of my repository anymore. I’d like to know if this is something which could be easily supported?

**Comment by natefinch** [comment]
It's not a good idea to have two modules in the same repo. It causes a lot of problems with go in general. You can do it, but it's kind of a pain. 

This might be fixable, but I'm not 100% sure yet.  Can you show me what kind of output mage produces in this case?

**Reply by yann-soubeyrand** [reply]
> It's not a good idea to have two modules in the same repo. It causes a lot of problems with go in general. You can do it, but it's kind of a pain.

Yes, I’m still weighting the pros and cons of this solution.

> This might be fixable, but I'm not 100% sure yet. Can you show me what kind of output mage produces in this case?

Here is a small test repo: https://github.com/yann-soubeyrand/nested-go-module-test. Running `mage` at its root gives me
```
Error parsing magefiles: error running "go list -f {{.Dir}}||{{.Name}} github.com/yann-soubeyrand/nested-go-module-test/magefiles/test": exit status 1
no required module provides package github.com/yann-soubeyrand/nested-go-module-test/magefiles/test; to add it:
	go get github.com/yann-soubeyrand/nested-go-module-test/magefiles/test
```

**Reply by natefinch** [reply, marked as answer]
Yeah, I think this is because mage is actually running in the root directory, so it's using the go.mod of the root directory. It probably shouldn't do that if you have a `magefiles` directory, instead it should change the current working directory to that directory to build the magefiles. 

I _think_ this is fixable, but I'm a little worried it may break other people. I'll have to look at it and do some testing.

**Reply by yann-soubeyrand** [reply]
Thanks!
